integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/NotPredicateSpecificsIT.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.tck.search.predicate;
+
+import static org.hibernate.search.util.impl.integrationtest.common.assertion.SearchResultAssert.assertThatQuery;
+
+import org.hibernate.search.engine.backend.document.IndexFieldReference;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaElement;
+import org.hibernate.search.engine.search.predicate.dsl.BooleanPredicateClausesStep;
+import org.hibernate.search.engine.search.predicate.dsl.SearchPredicateFactory;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.util.rule.SearchSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.BulkIndexer;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.SimpleMappedIndex;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class NotPredicateSpecificsIT {
```

Comment:
That's for the `not`-specific ITs, but you should also add something to cover base features. In this case I think there's only the score to worry about. See `OrPredicateBaseIT` for example (though you'll only need one nested test for the `not` predicate).

Reply:
right ... I didn't have a `NotPredicateBaseIT` as I made this not predicate to only have a final step, so there's no way to set a boost or anything. At that point, I was looking at this `not` as something only to negate a predicate. But I suppose, based on this, and your other comment from above, `not` should allow setting the boost/score too? Or I'm misinterpreting 😟 😃

Reply:
> I made this not predicate to only have a final step, so there's no way to set a boost or anything

Ah, I didn't notice that.

I think that a `not` predicate will always have a score of `1` when it matches, because then the inner predicate *did not* match and thus the score of the inner predicate is irrelevant. So, `boost`, at least, could make sense, to change that score of `1` to something else. `constantScore` isn't very useful here, but it's just simpler to also expose it.